pkg/devicemapper/devmapper.go

```diff
@@ -146,16 +150,31 @@ func TaskCreate(tasktype TaskType) *Task {
 	if Ctask == nil {
 		return nil
 	}
-	task := &Task{unmanaged: Ctask}
+	task := &Task{unmanaged: Ctask, timeout: defaultTimeout}
 	runtime.SetFinalizer(task, (*Task).destroy)
 	return task
 }
 
 func (t *Task) run() error {
-	if res := DmTaskRun(t.unmanaged); res != 1 {
-		return ErrTaskRun
+	ch := make(chan error)
```

Comment:
Does this need to be a buffered channel, so that the goroutine won't hang forever if the `select` statement hits the timeout and stops reading from the channel?